### PR TITLE
feat(smartWallet): defer deposits until purse available

### PR DESCRIPTION
**Pull request by turadg**
supporting #6157 

## Description

Some brands aren't available when a payments to the deposit facet. In this case we now hold the payments in a queue and when the purse is available we deposit all of them.

The payouts handler now uses the deposit facet to get the same feature. It also attempts each payout separately instead of the previous behavior of blocking until all payments were available.

### Security Considerations

Deposits are unpermissioned so queuing them opens an attack to grow the queue unbounded. The collections are durable which should mitigate RAM consumption. If there are more problems in practice the queue size can be capped. I considered doing that in this PR but considered a worse vulnerability is for an attacker to fill the queue of a wallet and prevent legitimate deposits.

### Documentation Considerations

--
### Testing Considerations

--

### CODE REVIEW

File: packages/smart-wallet/src/payments.js [hunk 1]
```diff
@@ -74,20 +75,14 @@ export const makePaymentsHelper = purseForBrand => {
...
      */
     async depositPayouts(payouts) {
-      /** @type {PaymentKeywordRecord} */
-      // @ts-expect-error ???
-      const paymentKeywordRecord = await deeplyFulfilledObject(payouts);
       /** Record<string, Promise<Amount>> */
-      const amountPKeywordRecord = objectMap(paymentKeywordRecord, payment =>
-        E(purseForBrand(payment.getAllegedBrand())).deposit(payment),
+      const amountPKeywordRecord = objectMap(payouts, paymentRef =>
+        E.when(paymentRef, payment => depositFacet.receive(payment)),
```
Comment (dckc) [inline packages/smart-wallet/src/payments.js:85]: :+1: good idea to re-use the depositFacet

File: packages/smart-wallet/src/smartWallet.js [hunk 2]
```diff
@@ -248,27 +259,46 @@ const behavior = {
...
-     * NB: the previous smart wallet contract would try again each time there's a new issuer.
-     * This version does not: 1) for expedience, 2: to avoid resource exhaustion vulnerability.
+     * @returns {Promise<Amount>} amounts for deferred deposits will be empty
      */
     async receive(payment) {
       /** @type {State} */
-      const { brandPurses } = this.state;
+      const { brandPurses, paymentQueues: queues } = this.state;
       const brand = await E(payment).getAllegedBrand();
       const purse = brandPurses.get(brand);
```
Comment (dckc) [inline packages/smart-wallet/src/smartWallet.js]: This get will throw rather than returning undefined, no?

**Review by dckc — CHANGES_REQUESTED** [review]
`get` throws on our collections, no?

(reusing the standard name with different behavior was... an odd choice, IMO)

File: packages/smart-wallet/src/smartWallet.js (hunk 2 above)
Comment (turadg) [reply packages/smart-wallet/src/smartWallet.js]: yes good catch! I added a unit test for regression and it caught a missing harden too.

**Review by dckc — APPROVED** [review]
thanks for the unit test

**Pull request merged**